tests/E2E/test/campaigns/common_scenarios/product.js

```diff
@@ -55,8 +55,9 @@ module.exports = {
       test('should click on "New Product" button', () => {
         return promise
           .then(() => client.waitForExistAndClick(AddProductPage.new_product_button))
+          .then(() => client.isVisible(AddProductPage.symfony_toolbar, 3000))
           .then(() => {
-            if (global.ps_mode_dev) {
+            if (global.ps_mode_dev && global.isVisible) {
```

Comment:
You maybe need to create an helper to remove this duplicate call. Like `client.waitForSymfonyToolbar(AddProductPage)`
and this function can contain the check about ps_mode_dev and isVisible and the client.waitForExsistAndClick :)